sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -370,9 +378,56 @@ static ActivityExtensions()
         private static Func<object, bool>? ActivitySourceHasListenersMethod;
         private static Func<string, string?, ICollection<KeyValuePair<string, object>>?, object?>? CreateActivityLinkMethod;
         private static Func<ICollection<KeyValuePair<string,object>>?>? CreateTagsCollectionMethod;
-
+        private static Func<Activity, string, object?>? GetCustomPropertyMethod;
+        private static Action<Activity, string, object>? SetCustomPropertyMethod;
+        private static string AzureSdkScopeLabel = "az.sdk.scope";
         private static readonly ParameterExpression ActivityParameter = Expression.Parameter(typeof(Activity));
 
+        public static bool IsNestedScopeEnabled(this Activity activity)
```

Comment:
We probably should define `GetCustomProperty` / `SetCustomProperty` pair of methods and have the bit that accesses `"az.sdk.scope"` property in `DiagnosticScope`.